frontend/src/components/pages/Default.tsx

```diff
@@ -40,8 +59,39 @@ const Default = (): React.ReactElement => {
     getPublicCourseData,
     allCourseData,
     publicCourseData,
+    getCourseProgresses,
+    courseProgressesData,
   ]);
 
+  const checkFilter = (tab_name: string, courseId: string): boolean => {
+    if (tab_name === "All Courses") {
+      return true;
+    }
+    if (!courseProgresses) {
+      return true;
+    }
+    const courseObj = courseProgresses.find(
+      (course) => course.courseId === courseId,
```

Comment:
This is kind of inefficient (quadratic time), which you could work around by making courseProgresses a map instead. However, the scale might be too small for this to matter so I'm wondering your thoughts.

Reply:
my original thought process was because the scale is pretty small it was okay, but I think ill refactor anyways